### ISSUE TITLE
[Android] Severe glitch when navigating to a screen with google_mobile_ads 

### ISSUE THREAD

**Issue by peterwvj** [flutter/flutter#95343]
Upgrading to Flutter 2.8.0 causes flickering when navigating between screens in apps that contain ads. In particular, the flickering happens when navigating to a new screen from another screen that has a banner ad. When commenting out the banner the flickering goes away.

The problem is not producible using Flutter 2.5.3.

The issue was originally raised in [googleads-mobile-flutter](https://github.com/googleads/googleads-mobile-flutter/issues/467) but after investigating the issue further it is now believed to be a problem with the Flutter SDK.

To produce the problem, run the attached application code using google_mobile_ads: 1.0.1 and Flutter 2.8.0.

Video sample, code sample and `flutter doctor -v` output produced by  @maheshmnj

<details>
<summary>Code sample</summary>

```dart

import 'dart:io';

import 'package:flutter/material.dart';
import 'package:google_mobile_ads/google_mobile_ads.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(home: HomeScreen());
  }
}



class HomeScreen extends StatefulWidget {
  const HomeScreen({Key? key}) : super(key: key);

  @override
  _HomeScreenState createState() => _HomeScreenState();
}

class _HomeScreenState extends State<HomeScreen> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        children: [
          Expanded(child: ListView.builder(itemBuilder: (context, index) {
            return ListTile(
              title: Text('Item $index'),
              onTap: () {
                Navigator.push(context, MaterialPageRoute(builder: (context) {
                  return const ProducDetailPage();
                }));
              },
            );
          })),
          const CustomBannerAd()
        ],
      ),
    );
  }
}

class ProducDetailPage extends StatelessWidget {
  const ProducDetailPage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: const Text('Produc Detail'),
        ),
        body: Column(children: const [
          SizedBox(
            height: 150,
            child: Placeholder(
              color: Colors.red,
            ),
          ),
          SizedBox(
            height: 150,
            child: Placeholder(
              color: Colors.green,
            ),
          ),
          CustomBannerAd()
        ]));
  }
}

class CustomBannerAd extends StatefulWidget {
  const CustomBannerAd({Key? key}) : super(key: key);

  @override
  _CustomBannerAdState createState() => _CustomBannerAdState();
}

class _CustomBannerAdState extends State<CustomBannerAd> {
  BannerAd? _anchoredAdaptiveAd;
  var _isLoaded = false;
  late Orientation _currentOrientation;

  @override
  void didChangeDependencies() {
    super.didChangeDependencies();

    _currentOrientation = MediaQuery.of(context).orientation;
    _loadAd();
  }

  @override
  void dispose() {
    super.dispose();

    _anchoredAdaptiveAd?.dispose();
  }

  Future<void> _loadAd() async {
    await _anchoredAdaptiveAd?.dispose();

    if (mounted) {
      setState(() {
        _anchoredAdaptiveAd = null;
        _isLoaded = false;
      });
    }

    final AnchoredAdaptiveBannerAdSize? size =
        await AdSize.getCurrentOrientationAnchoredAdaptiveBannerAdSize(
            MediaQuery.of(context).size.width.truncate());

    if (size == null) {
      debugPrint('Unable to get height of anchored banner.');
      return;
    }

    _anchoredAdaptiveAd = BannerAd(
      adUnitId: Platform.isAndroid
          ? 'ca-app-pub-3940256099942544/6300978111'
          : 'ca-app-pub-3940256099942544/2934735716',
      size: size,
      request: const AdRequest(
        nonPersonalizedAds: false,
      ),
      listener: BannerAdListener(
        onAdLoaded: (Ad ad) {
          debugPrint("Ad loaded");
          if (mounted) {
            setState(() {
              _anchoredAdaptiveAd = ad as BannerAd;
              _isLoaded = true;
            });
          }
        },
        onAdFailedToLoad: (Ad ad, LoadAdError error) {
          debugPrint('$BannerAd failedToLoad: $error');
          ad.dispose();
        },
        onAdOpened: (Ad ad) => debugPrint('$BannerAd onAdOpened.'),
        onAdClosed: (Ad ad) => debugPrint('$BannerAd onAdClosed.'),
      ),
    );

    return _anchoredAdaptiveAd?.load();
  }

  @override
  Widget build(BuildContext context) {
    return OrientationBuilder(builder: (context, orientation) {
      if (_currentOrientation == orientation &&
          _anchoredAdaptiveAd != null &&
          _isLoaded) {
        return Container(
          color: Colors.transparent,
          width: _anchoredAdaptiveAd!.size.width.toDouble(),
          height: _anchoredAdaptiveAd?.size.height.toDouble(),
          child: AdWidget(ad: _anchoredAdaptiveAd!),
        );
      }

      // Reload the ad if the orientation changes.
      if (_currentOrientation != orientation) {
        _currentOrientation = orientation;
        _loadAd();
      }
      return const SizedBox.shrink();
    });
  }
}
```
</details>

Video showing the bug (Flutter 2.8.0), This bug is not reproducible using Flutter 2.5.3:

stable 2.8.0 | stable 2.5.3
:-: | :-:
<video src='https://user-images.githubusercontent.com/5388919/146231131-d7ee8b4e-5457-4fca-83dd-964e2d6d2285.mov' width=180/> | <video src='https://user-images.githubusercontent.com/5388919/146231261-9c1dd620-f570-41a0-ac38-8a87c1075e6d.mov' width=180/>

Can presumably be reproduced using any Android 12 device. See `flutter doctor -v` output for details.

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 2.8.0, on macOS 12.0.1 21A559 darwin-arm, locale en-GB)
    • Flutter version 2.8.0 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cf44000065 (2 days ago), 2021-12-08 14:06:50 -0800
    • Engine revision 40a99c5951
    • Dart version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.61.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (3 available)
    • Redmi K20 Pro (mobile) • 192.168.1.6:5555 • android-arm64  • Android 11 (API 30)
    • macOS (desktop)        • macos            • darwin-arm64   • macOS 12.0.1 21A559 darwin-arm
    • Chrome (web)           • chrome           • web-javascript • Google Chrome 96.0.4664.93

• No issues found!
```
</details>

**Comment by DabPb** [comment]
Facing same issue!

**Comment by CripyIce** [comment]
Same here

**Comment by MagnusJohansson** [comment]
Same here.  Have published apps in productions stores with this defect, a bit annoying... 
It doesn't seem to affect iOS though, only Android.

**Comment by mustafa-707** [comment]
https://user-images.githubusercontent.com/74349843/146352418-25905653-d336-41b6-865e-587dcfe7103a.mp4

Same here : flutter 2.8.0 was not appear on 2.5.3

**Comment by peterwvj** [comment]
Yes, this is preventing me from publishing to Production as well. Unfortunately downgrading to 2.5.3 its not an option for me due to some other issue.

**Comment by maheshmnj** [comment]
Hi @peterwvj, Thanks for filing the issue. I am able to reproduce the issue with stable 2.8.0 and the `master, 2.9.0-1.0.pre.93`, But not on `stable 2.5.3`. This issue is reproducible only on Xiaomi k20 pro(Android 11) and works fine on IOS.
 
Here is a [timeline trace](https://github.com/flutter/flutter/files/7726508/ads_timeline_trace.json.zip) run using skia flag enabled on the stable channel, if this can help track down the bug.


 <details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 2.8.0, on macOS 12.0.1 21A559 darwin-arm, locale en-GB)
    • Flutter version 2.8.0 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cf44000065 (8 days ago), 2021-12-08 14:06:50 -0800
    • Engine revision 40a99c5951
    • Dart version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.61.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (4 available)
    • Redmi K20 Pro (mobile) • 192.168.1.6:5555                     • android-arm64  • Android 11 (API 30)
    • iPhone 11 (mobile)     • FC644E6C-7F02-40E5-9998-1F0CD162FBD8 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.0.1 21A559 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 96.0.4664.110

• No issues found!
```

```
[✓] Flutter (Channel master, 2.9.0-1.0.pre.93, on macOS 12.0.1 21A559 darwin-arm, locale en-GB)
    • Flutter version 2.9.0-1.0.pre.93 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 1be7f1ea56 (31 hours ago), 2021-12-14 22:14:07 -0500
    • Engine revision e444009bf4
    • Dart version 2.16.0 (build 2.16.0-109.0.dev)
    • DevTools version 2.9.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.61.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (4 available)
    • Redmi K20 Pro (mobile) • 192.168.1.6:5555                     • android-arm64  • Android 11 (API 30)
    • iPhone 11 (mobile)     • FC644E6C-7F02-40E5-9998-1F0CD162FBD8 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.0.1 21A559 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 96.0.4664.110

• No issues found!
mahesh@Maheshs-MacBoo
```

```
[✓] Flutter (Channel stable, 2.5.3, on macOS 12.0.1 21A559 darwin-arm, locale en-GB)
    • Flutter version 2.5.3 at /Users/mahesh/Documents/flutter_beta
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (9 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.1, Build version 13A1030d
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.61.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (4 available)
    • Redmi K20 Pro (mobile) • 192.168.1.6:5555                     • android-arm64  • Android 11 (API 30)
    • iPhone 11 (mobile)     • FC644E6C-7F02-40E5-9998-1F0CD162FBD8 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.0.1 21A559 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 96.0.4664.110

• No issues found!
```
</details>


<details>
<summary>logs in profile mode with trace skia</summary>

```
mahesh@Maheshs-MacBook-Air-M1 ads_example % flutter run --profile --trace-skia
Multiple devices found:
Redmi K20 Pro (mobile) • 192.168.1.6:5555                     • android-arm64  • Android 11 (API 30)
iPhone 11 (mobile)     • FC644E6C-7F02-40E5-9998-1F0CD162FBD8 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.0.1 21A559 darwin-arm
Chrome (web)           • chrome                               • web-javascript • Google Chrome 96.0.4664.110
[1]: Redmi K20 Pro (192.168.1.6:5555)
[2]: iPhone 11 (FC644E6C-7F02-40E5-9998-1F0CD162FBD8)
[3]: macOS (macos)
[4]: Chrome (chrome)
Please choose one (To quit, press "q/Q"): 1
Launching lib/main.dart on Redmi K20 Pro in profile mode...
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
Running Gradle task 'assembleProfile'...                           41.5s
✓  Built build/app/outputs/flutter-apk/app-profile.apk (25.7MB).
Installing build/app/outputs/flutter-apk/app.apk...                 7.8s
W/FlutterActivityAndFragmentDelegate(17880): A splash screen was provided to Flutter, but this is deprecated. See flutter.dev/go/android-splash-migration for migration steps.

Flutter run key commands.
h List all available interactive commands.
c Clear the screen
q Quit (terminate the application on the device).
An Observatory debugger and profiler on Redmi K20 Pro is available at: http://127.0.0.1:52114/2Dqa1-EPtco=/
W/Ads     (17880): Not retrying to fetch app settings
I/Ads     (17880): Ad failed to load : 3
I/flutter (17880): BannerAd failedToLoad: LoadAdError(code: 3, domain: com.google.android.gms.ads, message: Ad request doesn't meet size requirements. <https://support.google.com/admob/answer/99051
75#2>, responseInfo: ResponseInfo(responseId: null, mediationAdapterClassName: , adapterResponses: []))
The Flutter DevTools debugger and profiler on Redmi K20 Pro is available at: http://127.0.0.1:9101?uri=http://127.0.0.1:52114/2Dqa1-EPtco=/
D/EgretLoader(17880): EgretLoader(Context context)
D/EgretLoader(17880): The context is not activity
D/EgretLoader(17880): EgretLoader(Context context)
D/EgretLoader(17880): The context is not activity
W/ple.ads_exampl(17880): Accessing hidden method Landroid/media/AudioManager;->getOutputLatency(I)I (greylist, reflection, allowed)
W/cr_media(17880): Requires BLUETOOTH permission
I/DynamiteModule(17880): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:214106404
I/DynamiteModule(17880): Selected remote version of com.google.android.gms.ads.dynamite, version >= 214106404
I/flutter (17880): Ad loaded
W/ContentCatcher(17880): Failed to notify a WebView
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=129879496, downTime=129879496 } moveCount:0
W/MirrorManager(17880): this model don't Support
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=129879721, downTime=129879496 } moveCount:17
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=129880106, downTime=129880106 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=129880417, downTime=129880106 } moveCount:25
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=129881031, downTime=129881031 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=129881275, downTime=129881031 } moveCount:16
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=129882532, downTime=129882532 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=129882977, downTime=129882532 } moveCount:35
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=129883372, downTime=129883372 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=129883640, downTime=129883372 } moveCount:23
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=129884625, downTime=129884625 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=129884759, downTime=129884625 } moveCount:0
D/DynamitePackage(17880): Instantiating com.google.android.gms.ads.ChimeraAdManagerCreatorImpl
I/Ads     (17880): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7533D24B76F3D928E20EB84A8C420B39")) to get test ads on this device.
I/AudioManager(17880): In isSpeakerphoneOn(), calling application: com.example.ads_example
D/EgretLoader(17880): EgretLoader(Context context)
D/EgretLoader(17880): The context is not activity
I/DynamiteModule(17880): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:214106404
I/DynamiteModule(17880): Selected remote version of com.google.android.gms.ads.dynamite, version >= 214106404
I/flutter (17880): Ad loaded
W/ContentCatcher(17880): Failed to notify a WebView
W/System  (17880): A resource failed to call release. 
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=129928691, downTime=129928691 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=129928806, downTime=129928691 } moveCount:0
W/ContentCatcher(17880): Failed to notify a WebView
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/chatty  (17880): uid=10919(com.example.ads_example) identical 3 lines
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=129932270, downTime=129932270 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=129932402, downTime=129932270 } moveCount:0
D/DynamitePackage(17880): Instantiating com.google.android.gms.ads.ChimeraAdManagerCreatorImpl
I/Ads     (17880): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7533D24B76F3D928E20EB84A8C420B39")) to get test ads on this device.
I/AudioManager(17880): In isSpeakerphoneOn(), calling application: com.example.ads_example
D/EgretLoader(17880): EgretLoader(Context context)
D/EgretLoader(17880): The context is not activity
I/DynamiteModule(17880): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:214106404
I/DynamiteModule(17880): Selected remote version of com.google.android.gms.ads.dynamite, version >= 214106404
I/flutter (17880): Ad loaded
W/ContentCatcher(17880): Failed to notify a WebView
W/System  (17880): A resource failed to call release. 
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=129941855, downTime=129941855 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=129941971, downTime=129941855 } moveCount:0
W/ContentCatcher(17880): Failed to notify a WebView
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/chatty  (17880): uid=10919(com.example.ads_example) identical 3 lines
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=129943322, downTime=129943322 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=129943456, downTime=129943322 } moveCount:0
D/DynamitePackage(17880): Instantiating com.google.android.gms.ads.ChimeraAdManagerCreatorImpl
I/Ads     (17880): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7533D24B76F3D928E20EB84A8C420B39")) to get test ads on this device.
I/AudioManager(17880): In isSpeakerphoneOn(), calling application: com.example.ads_example
D/EgretLoader(17880): EgretLoader(Context context)
D/EgretLoader(17880): The context is not activity
I/DynamiteModule(17880): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:214106404
I/DynamiteModule(17880): Selected remote version of com.google.android.gms.ads.dynamite, version >= 214106404
I/flutter (17880): Ad loaded
W/ContentCatcher(17880): Failed to notify a WebView
W/System  (17880): A resource failed to call release. 
I/Ads     (17880): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7533D24B76F3D928E20EB84A8C420B39")) to get test ads on this device.
I/AudioManager(17880): In isSpeakerphoneOn(), calling application: com.example.ads_example
D/EgretLoader(17880): EgretLoader(Context context)
D/EgretLoader(17880): The context is not activity
I/DynamiteModule(17880): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:214106404
I/DynamiteModule(17880): Selected remote version of com.google.android.gms.ads.dynamite, version >= 214106404
W/ContentCatcher(17880): Failed to notify a WebView
W/ContentCatcher(17880): Failed to notify a WebView
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/chatty  (17880): uid=10919(com.example.ads_example) identical 1 line
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/flutter (17880): Ad loaded
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130062925, downTime=130062925 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130063015, downTime=130062925 } moveCount:0
W/ContentCatcher(17880): Failed to notify a WebView
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/chatty  (17880): uid=10919(com.example.ads_example) identical 4 lines
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130063577, downTime=130063577 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130063778, downTime=130063577 } moveCount:14
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130064264, downTime=130064264 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130064407, downTime=130064264 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130064827, downTime=130064827 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130064918, downTime=130064827 } moveCount:0
D/DynamitePackage(17880): Instantiating com.google.android.gms.ads.ChimeraAdManagerCreatorImpl
I/Ads     (17880): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7533D24B76F3D928E20EB84A8C420B39")) to get test ads on this device.
I/AudioManager(17880): In isSpeakerphoneOn(), calling application: com.example.ads_example
D/EgretLoader(17880): EgretLoader(Context context)
D/EgretLoader(17880): The context is not activity
I/DynamiteModule(17880): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:214106404
I/DynamiteModule(17880): Selected remote version of com.google.android.gms.ads.dynamite, version >= 214106404
I/flutter (17880): Ad loaded
W/ContentCatcher(17880): Failed to notify a WebView
W/System  (17880): A resource failed to call release. 
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130066299, downTime=130066299 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130066415, downTime=130066299 } moveCount:0
W/ContentCatcher(17880): Failed to notify a WebView
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/chatty  (17880): uid=10919(com.example.ads_example) identical 3 lines
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/Ads     (17880): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7533D24B76F3D928E20EB84A8C420B39")) to get test ads on this device.
I/AudioManager(17880): In isSpeakerphoneOn(), calling application: com.example.ads_example
D/EgretLoader(17880): EgretLoader(Context context)
D/EgretLoader(17880): The context is not activity
I/DynamiteModule(17880): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:214106404
I/DynamiteModule(17880): Selected remote version of com.google.android.gms.ads.dynamite, version >= 214106404
W/ContentCatcher(17880): Failed to notify a WebView
W/ContentCatcher(17880): Failed to notify a WebView
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/chatty  (17880): uid=10919(com.example.ads_example) identical 1 line
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/flutter (17880): Ad loaded
I/Ads     (17880): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7533D24B76F3D928E20EB84A8C420B39")) to get test ads on this device.
I/AudioManager(17880): In isSpeakerphoneOn(), calling application: com.example.ads_example
D/EgretLoader(17880): EgretLoader(Context context)
D/EgretLoader(17880): The context is not activity
I/DynamiteModule(17880): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:214106404
I/DynamiteModule(17880): Selected remote version of com.google.android.gms.ads.dynamite, version >= 214106404
W/ContentCatcher(17880): Failed to notify a WebView
W/ContentCatcher(17880): Failed to notify a WebView
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/chatty  (17880): uid=10919(com.example.ads_example) identical 1 line
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/flutter (17880): Ad loaded
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130153114, downTime=130153114 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130153340, downTime=130153114 } moveCount:19
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130153953, downTime=130153953 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130154129, downTime=130153953 } moveCount:13
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130155121, downTime=130155121 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130155265, downTime=130155121 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130155866, downTime=130155866 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130155921, downTime=130155866 } moveCount:0
D/DynamitePackage(17880): Instantiating com.google.android.gms.ads.ChimeraAdManagerCreatorImpl
I/Ads     (17880): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7533D24B76F3D928E20EB84A8C420B39")) to get test ads on this device.
I/AudioManager(17880): In isSpeakerphoneOn(), calling application: com.example.ads_example
D/EgretLoader(17880): EgretLoader(Context context)
D/EgretLoader(17880): The context is not activity
I/DynamiteModule(17880): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:214106404
I/DynamiteModule(17880): Selected remote version of com.google.android.gms.ads.dynamite, version >= 214106404
I/flutter (17880): Ad loaded
W/ContentCatcher(17880): Failed to notify a WebView
W/System  (17880): A resource failed to call release. 
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130157192, downTime=130157192 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130157275, downTime=130157192 } moveCount:0
W/ContentCatcher(17880): Failed to notify a WebView
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/chatty  (17880): uid=10919(com.example.ads_example) identical 3 lines
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130157814, downTime=130157814 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130157989, downTime=130157814 } moveCount:13
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130158206, downTime=130158206 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130158442, downTime=130158206 } moveCount:1
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130158688, downTime=130158688 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130158754, downTime=130158688 } moveCount:0
D/DynamitePackage(17880): Instantiating com.google.android.gms.ads.ChimeraAdManagerCreatorImpl
I/Ads     (17880): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7533D24B76F3D928E20EB84A8C420B39")) to get test ads on this device.
I/AudioManager(17880): In isSpeakerphoneOn(), calling application: com.example.ads_example
W/Ads     (17880): Not retrying to fetch app settings
D/EgretLoader(17880): EgretLoader(Context context)
D/EgretLoader(17880): The context is not activity
I/DynamiteModule(17880): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:214106404
I/DynamiteModule(17880): Selected remote version of com.google.android.gms.ads.dynamite, version >= 214106404
I/flutter (17880): Ad loaded
W/ContentCatcher(17880): Failed to notify a WebView
W/System  (17880): A resource failed to call release. 
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130159952, downTime=130159952 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130160027, downTime=130159952 } moveCount:0
W/ContentCatcher(17880): Failed to notify a WebView
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/chatty  (17880): uid=10919(com.example.ads_example) identical 3 lines
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130160472, downTime=130160472 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130160672, downTime=130160472 } moveCount:15
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130160965, downTime=130160965 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130161217, downTime=130160965 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=130161510, downTime=130161510 } moveCount:0
D/MIUIInput(17880): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=130161584, downTime=130161510 } moveCount:0
D/DynamitePackage(17880): Instantiating com.google.android.gms.ads.ChimeraAdManagerCreatorImpl
I/Ads     (17880): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7533D24B76F3D928E20EB84A8C420B39")) to get test ads on this device.
I/AudioManager(17880): In isSpeakerphoneOn(), calling application: com.example.ads_example
D/EgretLoader(17880): EgretLoader(Context context)
D/EgretLoader(17880): The context is not activity
I/DynamiteModule(17880): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:214106404
I/DynamiteModule(17880): Selected remote version of com.google.android.gms.ads.dynamite, version >= 214106404
I/flutter (17880): Ad loaded
W/ContentCatcher(17880): Failed to notify a WebView
W/System  (17880): A resource failed to call release. 
I/Ads     (17880): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7533D24B76F3D928E20EB84A8C420B39")) to get test ads on this device.
I/AudioManager(17880): In isSpeakerphoneOn(), calling application: com.example.ads_example
D/EgretLoader(17880): EgretLoader(Context context)
D/EgretLoader(17880): The context is not activity
I/DynamiteModule(17880): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:214106404
I/DynamiteModule(17880): Selected remote version of com.google.android.gms.ads.dynamite, version >= 214106404
W/ContentCatcher(17880): Failed to notify a WebView
W/ContentCatcher(17880): Failed to notify a WebView
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/chatty  (17880): uid=10919(com.example.ads_example) identical 1 line
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/flutter (17880): Ad loaded
I/Ads     (17880): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7533D24B76F3D928E20EB84A8C420B39")) to get test ads on this device.
I/AudioManager(17880): In isSpeakerphoneOn(), calling application: com.example.ads_example
D/EgretLoader(17880): EgretLoader(Context context)
D/EgretLoader(17880): The context is not activity
I/DynamiteModule(17880): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:214106404
I/DynamiteModule(17880): Selected remote version of com.google.android.gms.ads.dynamite, version >= 214106404
W/ContentCatcher(17880): Failed to notify a WebView
W/ContentCatcher(17880): Failed to notify a WebView
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/chatty  (17880): uid=10919(com.example.ads_example) identical 1 line
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/flutter (17880): Ad loaded
I/Ads     (17880): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7533D24B76F3D928E20EB84A8C420B39")) to get test ads on this device.
I/AudioManager(17880): In isSpeakerphoneOn(), calling application: com.example.ads_example
I/ple.ads_exampl(17880): Waiting for a blocking GC ProfileSaver
I/ple.ads_exampl(17880): WaitForGcToComplete blocked ProfileSaver on ProfileSaver for 13.643ms
D/EgretLoader(17880): EgretLoader(Context context)
D/EgretLoader(17880): The context is not activity
I/DynamiteModule(17880): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:214106404
I/DynamiteModule(17880): Selected remote version of com.google.android.gms.ads.dynamite, version >= 214106404
W/ContentCatcher(17880): Failed to notify a WebView
W/ContentCatcher(17880): Failed to notify a WebView
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/chatty  (17880): uid=10919(com.example.ads_example) identical 1 line
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/flutter (17880): Ad loaded
I/Ads     (17880): Use RequestConfiguration.Builder().setTestDeviceIds(Arrays.asList("7533D24B76F3D928E20EB84A8C420B39")) to get test ads on this device.
I/AudioManager(17880): In isSpeakerphoneOn(), calling application: com.example.ads_example
D/EgretLoader(17880): EgretLoader(Context context)
D/EgretLoader(17880): The context is not activity
I/DynamiteModule(17880): Considering local module com.google.android.gms.ads.dynamite:0 and remote module com.google.android.gms.ads.dynamite:214106404
I/DynamiteModule(17880): Selected remote version of com.google.android.gms.ads.dynamite, version >= 214106404
W/ContentCatcher(17880): Failed to notify a WebView
W/ContentCatcher(17880): Failed to notify a WebView
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/chatty  (17880): uid=10919(com.example.ads_example) identical 1 line
W/Ads     (17880): #004 The webview is destroyed. Ignoring action.
I/flutter (17880): Ad loaded

```
</details>

cc: @blasten

**Comment by CodingWithTashi** [comment]
Heard flutter 2.8 has improve in terms of app performance and just upgrated to latest version and Now I am seeing glitch issue while navigating(Banner Ads). Funny things is that as long as you have ads in both screen ie. it work fine. It seems like I have to add ads in all screen to make it work otherwise downgrade the flutter version. 
Please check and update.

**Comment by peterwvj** [comment]
@blasten did you get a chance to look into this? Thanks in advance.

**Comment by westlinkin** [comment]
I have the same issue with NativeAd (google_mobile_ads 1.0.1) and Flutter (2.8.0)

**Comment by peterwvj** [comment]
I'm interested in hearing about workarounds in case someone knows some.

**Comment by asifuzzaman-reyad** [comment]
I have the same issue same problem with banner ads (google_mobile_ads 1.0.1) and Flutter (2.8.1)

**Comment by mustafa-707** [comment]
@asifuzzaman-reyad  i guess as @maheshmnj  mentioned above it will be at flutter 2.9.0

> Hi @peterwvj, Thanks for filing the issue. I am able to reproduce the issue with stable 2.8.0 and the `master, 2.9.0-1.0.pre.93`, But not on `stable 2.5.3`. This issue is reproducible only on Android and works fine on IOS.
> 
>  Screen.Recording.2021-12-16.at.4.18.17.PM.mov 
> Here is a [timeline trace](https://github.com/flutter/flutter/files/7726508/ads_timeline_trace.json.zip) run using skia flag enabled on the stable channel on Xiaomi k20 pro(Android 11), if this can help track down the bug.
> 
> flutter doctor -v
> logs in profile mode with trace skia
> Edit: This issue is only reproducible when using the ads plugin on `stable 2.8.0` and works fine on `stable 2.5.3`.
> 
> cc: @blasten

**Comment by jason-simmons** [comment]
This started with https://github.com/flutter/engine/commit/866db6211126ab34539b799852b9b63b9249c082

During the transition between the screens there may be a `PlatformViewsController.onEndFrame` callback where the `currentFrameUsedPlatformViewIds` set briefly becomes empty.

This will result in a call to `FlutterView.revertImageView` which will eventually call `FlutterRenderer.startRenderingToSurface`.  `startRenderingToSurface` will do a `Rasterizer::Teardown` before restarting the rasterizer for the new surface.

The `Rasterizer::Teardown` then does the `SurfacePool::DestroyLayers` call added in https://github.com/flutter/engine/commit/866db6211126ab34539b799852b9b63b9249c082, causing the flickering.

@blasten

**Comment by muhammadanas759** [comment]
I am also facing same issue when switch from 2.2.1 to 2.8.1.. it is very awkward issue. i have spend whole to find the cause but didn't find anything...

**Comment by westlinkin** [comment]
Do we have a solution or find a way to bypass this issue?

**Comment by AgentNemo** [comment]
Same issue

**Comment by q384264619** [comment]
face the same issue!

**Comment by q384264619** [comment]
> This started with [flutter/engine@866db62](https://github.com/flutter/engine/commit/866db6211126ab34539b799852b9b63b9249c082)
> 
> During the transition between the screens there may be a `PlatformViewsController.onEndFrame` callback where the `currentFrameUsedPlatformViewIds` set briefly becomes empty.
> 
> This will result in a call to `FlutterView.revertImageView` which will eventually call `FlutterRenderer.startRenderingToSurface`. `startRenderingToSurface` will do a `Rasterizer::Teardown` before restarting the rasterizer for the new surface.
> 
> The `Rasterizer::Teardown` then does the `SurfacePool::DestroyLayers` call added in [flutter/engine@866db62](https://github.com/flutter/engine/commit/866db6211126ab34539b799852b9b63b9249c082), causing the flickering.
> 
> @blasten

Do you have a solution or find a way to bypass this issue? @blasten

**Comment by maheshmnj** [comment]
@q384264619, Please don't tag team members repetitively. The issue has been prioritized and the issue seems to be [identified](https://github.com/flutter/flutter/issues/95343#issuecomment-999043140), The relevant team members will update on this issue when they have an update to share.

**Comment by NikhilMJain** [comment]
Following!

**Comment by SirJager** [comment]
Facing same issue in Flutter 2.8.1

https://user-images.githubusercontent.com/18713058/147820454-70582d6b-133a-45f1-b163-2b2286ee264f.mp4

**Comment by anasboyka** [comment]
> Heard flutter 2.8 has improve in terms of app performance and just upgrated to latest version and Now I am seeing glitch issue while navigating(Banner Ads). Funny things is that as long as you have ads in both screen ie. it work fine. It seems like I have to add ads in all screen to make it work otherwise downgrade the flutter version.
> Please check and update.

yeah thats the only way for me too... need to create another banner ads on the new page... guess thats the only way for now.. hope it got resolved..

**Comment by beisong7** [comment]
I am having the same issue. and it was not there as of the last build version but now it is frustrating users eyes on the new version.

**Comment by Rushil10** [comment]
Same Issue

**Comment by zanderso** [comment]
From triage: @blasten I'm bumping the priority on this since I think we may want to have a fix if possible before we cut the next branch.

**Comment by westlinkin** [comment]
Will this fix release to the stable version?

**Comment by blasten** [comment]
> Will this fix release to the stable version?

Yes, the `cp:` tags stand for "cherry-pick" into the specified version

**Comment by westlinkin** [comment]
> > Will this fix release to the stable version?
> 
> Yes, the `cp:` tags stand for "cherry-pick" into the specified version

Will this fix release as a 2.8.x patch? seems this is merged into 2.10

**Comment by zanderso** [comment]
@westlinkin No, sorry, we're in a shorter than usual release cycle, and this change has come in too late for a 2.8 hotfix. But don't worry, 2.10 is coming soon.

**Comment by chetanxpatil** [comment]
when are we getting the update still waiting

**Comment by maheshmnj** [comment]
This issue is fixed, and the fix is available on the latest master. You may either wait until the fix lands to stable in the upcoming release `stable 2.10`  or switch to the master channel using 

```
flutter channel master
flutter upgrade
```
**verified on  master, 2.9.0-1.0.pre.328**

https://user-images.githubusercontent.com/31410839/149135677-a5eb14c4-47e5-43ea-b4e6-e994d633bcaa.mov

**Comment by just-kip** [comment]
Issue is still reproducible on 2.10.0-0.2.pre. But not every time. I can see flickering once per 5 - 10 pushes.
So I have screens **A, B, C**. Screen **A** contains Native view. 

Push from A to B
Pop back from B to A
Push from A to C <--- In this case sometimes during the animation I see B screen for a while, but B is already popped.

I guess in original issue the problem is same.

Push from List to GoogleAds
Pop back from GoogleAds to List
Push from List to a new GoogleAds and during this animation old one is flickering.

I will try to do example a bit later.

**Comment by just-kip** [comment]
Please check first seconds of the video.

https://user-images.githubusercontent.com/4008338/150959900-1456af81-8597-4bd1-a4f8-0e1eca906366.mp4


Step to reproduce
- create new flutter project
- add `webview_flutter: ^3.0.0`
- copy code below to main.dart

<details>
<summary>main.dart</summary>

```dart
import 'package:flutter/material.dart';
import 'package:webview_flutter/webview_flutter.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(home: HomeScreen());
  }
}

class HomeScreen extends StatefulWidget {
  const HomeScreen({Key? key}) : super(key: key);

  @override
  _HomeScreenState createState() => _HomeScreenState();
}

class _HomeScreenState extends State<HomeScreen> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: Column(
          children: [
            ListTile(
              title: Text('Red'),
              onTap: () {
                Navigator.push(context, MaterialPageRoute(builder: (context) {
                  return const ProducDetailPage();
                }));
              },
            ),
            ListTile(
              title: Text('Green'),
              onTap: () {
                Navigator.push(context, MaterialPageRoute(builder: (context) {
                  return const ProductDetailPage2();
                }));
              },
            ),
            SizedBox(
              height: 150,
              child: WebView(
                initialUrl: 'https://flutter.dev',
              ),
            ),
          ],
        ),
      ),
    );
  }
}

class ProducDetailPage extends StatelessWidget {
  const ProducDetailPage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: const Text('Produc Detail'),
        ),
        body: Column(children: const [
          Padding(
            padding: EdgeInsets.all(8.0),
            child: SizedBox(
              height: 150,
              child: Placeholder(
                color: Colors.red,
              ),
            ),
          ),
        ]));
  }
}

class ProductDetailPage2 extends StatelessWidget {
  const ProductDetailPage2({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: const Text('Produc Detail'),
        ),
        body: Column(children: const [
          SizedBox(
            height: 150,
            child: Placeholder(
              color: Colors.green,
            ),
          ),
        ]));
  }
}import 'package:flutter/material.dart';
import 'package:webview_flutter/webview_flutter.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(home: HomeScreen());
  }
}

class HomeScreen extends StatefulWidget {
  const HomeScreen({Key? key}) : super(key: key);

  @override
  _HomeScreenState createState() => _HomeScreenState();
}

class _HomeScreenState extends State<HomeScreen> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: Column(
          children: [
            ListTile(
              title: Text('Red'),
              onTap: () {
                Navigator.push(context, MaterialPageRoute(builder: (context) {
                  return const ProducDetailPage();
                }));
              },
            ),
            ListTile(
              title: Text('Green'),
              onTap: () {
                Navigator.push(context, MaterialPageRoute(builder: (context) {
                  return const ProductDetailPage2();
                }));
              },
            ),
            SizedBox(
              height: 150,
              child: WebView(
                initialUrl: 'https://flutter.dev',
              ),
            ),
          ],
        ),
      ),
    );
  }
}

class ProducDetailPage extends StatelessWidget {
  const ProducDetailPage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: const Text('Produc Detail'),
        ),
        body: Column(children: const [
          Padding(
            padding: EdgeInsets.all(8.0),
            child: SizedBox(
              height: 150,
              child: Placeholder(
                color: Colors.red,
              ),
            ),
          ),
        ]));
  }
}

class ProductDetailPage2 extends StatelessWidget {
  const ProductDetailPage2({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: const Text('Produc Detail'),
        ),
        body: Column(children: const [
          SizedBox(
            height: 150,
            child: Placeholder(
              color: Colors.green,
            ),
          ),
        ]));
  }
}

```
</details>

```
➜  ~ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, 2.10.0-0.2.pre, on macOS 12.0.1 21A559 darwin-x64, locale en-RU)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[✓] VS Code (version 1.63.2)
[✓] Connected device (4 available)
[✓] HTTP Host Availability

• No issues found!
```

**Comment by zanderso** [comment]
@blasten ptal.

**Comment by blasten** [comment]
@just-kip That doesn't seem quite right. I am taking a closer look

**Comment by maheshmnj** [comment]
I verified the latest master based on the above sample code in the original post but the issue was not reproducible

https://user-images.githubusercontent.com/31410839/151121086-e46158b1-eb2c-4e9e-a752-162aa93aca34.mov

I tried using web view in place of google mobile ads and I did see the flicker and the app crashed. The issue has been filed here for further tracking https://github.com/flutter/flutter/issues/97283 

https://user-images.githubusercontent.com/31410839/151123465-f39ecd14-db6b-4447-9b41-89574b03db95.mov


<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 2.9.0-1.0.pre.504, on macOS 12.1 21C52 darwin-arm, locale en-GB)
    • Flutter version 2.9.0-1.0.pre.504 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 94d9b02deb (4 hours ago), 2022-01-25 22:00:07 -0600
    • Engine revision 54e1f3f4bb
    • Dart version 2.17.0 (build 2.17.0-54.0.dev)
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.32.0

[✓] Connected device (3 available)
    • Redmi K20 Pro (mobile) • 192.168.1.3:5555 • android-arm64  • Android 11 (API 30)
    • macOS (desktop)        • macos            • darwin-arm64   • macOS 12.1 21C52 darwin-arm
    • Chrome (web)           • chrome           • web-javascript • Google Chrome 97.0.4692.99

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by faizahmaddae** [comment]
> Upgrading to Flutter 2.8.0 causes flickering when navigating between screens in apps that contain ads. In particular, the flickering happens when navigating to a new screen from another screen that has a banner ad. When commenting out the banner the flickering goes away.
> 
> The problem is not producible using Flutter 2.5.3.
> 
> The issue was originally raised in [googleads-mobile-flutter](https://github.com/googleads/googleads-mobile-flutter/issues/467) but after investigating the issue further it is now believed to be a problem with the Flutter SDK.
> 
> To produce the problem, run the attached application code using google_mobile_ads: 1.0.1 and Flutter 2.8.0.
> 
> Video sample, code sample and `flutter doctor -v` output produced by @maheshmnj
> 
> Code sample
> Video showing the bug (Flutter 2.8.0), This bug is not reproducible using Flutter 2.5.3:
> 
> stable 2.8.0	stable 2.5.3
>  flutter-2.8.0-bug.mov 
>  flutter-2.5.3-no-bug.mov 
> Can presumably be reproduced using any Android 12 device. See `flutter doctor -v` output for details.
> 
> flutter doctor -v

i also have this glitch, can some help me ?

**Comment by blasten** [comment]
@maheshmnj the crash in #97283 is a different issue due to https://github.com/flutter/flutter/issues/97283#issuecomment-1022441333.

The glitch is a regression.

**Comment by faizahmaddae** [comment]
How solved is it?

**Comment by salpers** [comment]
I am observing the same behaviour when pushing a new Scaffold on top of a Google Maps widget.

**Comment by faizahmaddae** [comment]
is there any solution ??

**Comment by medst** [comment]
two month and the issue is not fixed yet !!!

**Comment by chitgoks** [comment]
> two month and the issue is not fixed yet !!!

it is already in the master channel, if you read the thread. and its going to be part of flutter 2.10 stable.

**Comment by chitgoks** [comment]
> two month and the issue is not fixed yet !!!

it is already in the master channel, if you read the thread. and its going to be part of flutter 2.10 stable.

**Comment by blasten** [comment]
I tested fdd0af78bbda27e1084ec859b27765d927cbe27e (Flutter 2.10-beta) on a Pixel 4, and I'm unable to reproduce the glitch.

Examples tried:

@just-kip https://github.com/flutter/flutter/issues/95343#issuecomment-1021037818

@peterwvj https://github.com/flutter/flutter/issues/95343#issue-1081292044


If you are still seeing this issue, please ensure that you are using that commit as the baseline. If you are still seeing this issue, this might be a device-specific issue. Please share the device model as well as the example code.

**Comment by MilesAdamson** [comment]
I have upgraded to ` 2.10.0-0.3.pre` and this does not appear to be fixed at all. I can also reproduce on my physical device in release mode.

https://user-images.githubusercontent.com/29047682/151905152-ab4a3804-74b5-45d7-aec6-bc00a287c066.mp4

**Comment by westlinkin** [comment]
The latest 2.10 version still got this problem.  For first time only though. pop the second screen and push again, the blink goes away.

**Comment by CodingWithTashi** [comment]
Just noticed after deployed app to production, It is bit annoying too. Thought issue was fixed, Please look into it.

**Comment by marcellocamara** [comment]
This problem was much worse in stable version 2.8.1
On the new flutter stable version 2.10.0, with dart 2.16.0, is much better. But it is **not fixed** 100% yet.
I can also see the glitch 😢

**Comment by CarGuo** [comment]
I test on the 2.10 stable version and the issue was reproducible sometime , look like thee video below: 


https://user-images.githubusercontent.com/10770362/152925664-e92d7e70-2abe-41dd-9d8b-6a3060b03404.mp4


```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel unknown, 2.10.0, on macOS 12.1 21C52 darwin-x64, locale zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.1)
[✓] IntelliJ IDEA Ultimate Edition (version 2018.3.5)
[!] Proxy Configuration
    ! NO_PROXY is not set
[✓] Connected device (2 available)
[✓] HTTP Host Availability

! Doctor found issues in 1 category.
```

**Comment by westlinkin** [comment]
it's almost two month, I am still using version 2.5.3 to build android app.

**Comment by blasten** [comment]
I modified the example a little bit, and I can see the issue now. I'm taking a look at the issue.

<details>
<summary>Code</summary>

```dart
void main() {
  WebView.platform = SurfaceAndroidWebView();
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(home: HomeScreen());
  }
}

class HomeScreen extends StatefulWidget {
  const HomeScreen({Key? key}) : super(key: key);

  @override
  _HomeScreenState createState() => _HomeScreenState();
}

class _HomeScreenState extends State<HomeScreen> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: Column(
          children: [
            ListTile(
              title: Text('Red'),
              onTap: () {
                Navigator.push(context, MaterialPageRoute(builder: (context) {
                  return const ProducDetailPage();
                }));
              },
            ),
            ListTile(
              title: Text('Green'),
              onTap: () {
                Navigator.push(context, MaterialPageRoute(builder: (context) {
                  return const ProductDetailPage2();
                }));
              },
            ),
            SizedBox(
              height: 150,
              child: WebView(
                initialUrl: 'https://flutter.dev',
              ),
            ),
          ],
        ),
      ),
    );
  }
}

class ProducDetailPage extends StatelessWidget {
  const ProducDetailPage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: const Text('Produc Detail'),
        ),
        body: Column(children: const [
          Padding(
            padding: EdgeInsets.all(8.0),
            child: SizedBox(
              height: 150,
              child: Placeholder(
                color: Colors.red,
              ),
            ),
          ),
          SizedBox(
            height: 150,
            child: WebView(
              initialUrl: 'https://flutter.dev',
            ),
          ),
        ]));
  }
}

class ProductDetailPage2 extends StatelessWidget {
  const ProductDetailPage2({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: const Text('Produc Detail'),
        ),
        body: Column(children: const [
          SizedBox(
            height: 150,
            child: Placeholder(
              color: Colors.green,
            ),
          ),
        ]));
  }
}
```
</details>

**Comment by blasten** [comment]
I synced to the commit prior to https://github.com/flutter/engine/commit/866db6211126ab34539b799852b9b63b9249c082, and I'm still able to see a glitch if I run the code snippet I shared above.

The good news is that once https://github.com/flutter/engine/pull/31198 is merged, this issue will be fixed from its root cause and won't happen again.

**Comment by shahoit** [comment]
From iraq 
, now im test flutter google_ads that was have crash when tap to other page but now fixed in google_mobile_ads:1.1.0
with flutter version 2.10.1 stabel versiom now the erorr not happening  screen not crashing anymore  if you have this error  contact me i will help you
Email:sh4hoit@gmail.com
Snapchat:shaho-it
Instagram:shaho_it

**Comment by westlinkin** [comment]
@blasten Will this be released in 2.10 hot fix?

**Comment by chitgoks** [comment]
i dont have a problem with flutter 2.10 and google ads 1.1.0 anymore.

**Comment by mustafa-707** [comment]
` flutter 2.10` and `google ads ^1.1.0 `. and it show some time the flicking

**Comment by railsonsm** [comment]
flutter 2.10 and google_mobile_ads ^1.1.0 . It seems to me that the problem is still occurring. Right when I updated it stopped blinking but soon after it came back

**Comment by lhdio** [comment]
flutter 2.10.2 and google_mobile_ads: ^1.1.0 . It seems to me that the problem is still occurring. :(

**Comment by Waheed388** [comment]
Flutter 2.10.2, 
Dart 2.16.1,
google_mobile_ads: ^1.1.0,

Such a turnoff. Cannot release the app to the play store because of this issue.

**Comment by lhdio** [comment]
flutter 2.10.3 and google_mobile_ads: ^1.1.0 . It seems to me that the problem is still occurring. :(

**Comment by CarGuo** [comment]
> I'm testing in the 2.10 stable version and the issue was reproducible sometime , look like thee video below:
> 
>  Screenrecorder-2022-02-08-13-40-48-945.mp4 
> 
> ```
> Doctor summary (to see all details, run flutter doctor -v):
> [✓] Flutter (Channel unknown, 2.10.0, on macOS 12.1 21C52 darwin-x64, locale zh-Hans-CN)
> [✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
> [✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
> [✓] Chrome - develop for the web
> [✓] Android Studio (version 2021.1)
> [✓] IntelliJ IDEA Ultimate Edition (version 2018.3.5)
> [!] Proxy Configuration
>     ! NO_PROXY is not set
> [✓] Connected device (2 available)
> [✓] HTTP Host Availability
> 
> ! Doctor found issues in 1 category.
> ```

2.10.3 still occurring , in my case with webView_flutter ：

1、`A page` -> `webview page` -> `B page` ， when `webview page` open `B page` ，sometime `A page` UI  will  twinkle in  `B page `.
2、when `B page` back to `webview page `, then back to `A page`， sometime `B page` UI  will  twinkle in `A page`  suddenly.

**if I only `A page` -> `webview page` , I can resolve this issue by render Container replace WebView when Navigator.pop .**

**Comment by mchlsndrn** [comment]
Any progress on this? Until this gets fixed I'm stuck on 2.5.3 and with me probably most people using the ads package.
Multiple users have confirmed this still happens with 2.10.3.
Apparently support for ad-based apps is not a priority?

**Comment by mustafa-707** [comment]
same here stuck on ` 2.5.3`

**Comment by zanderso** [comment]
Hi folks, I believe the [fix](https://github.com/flutter/engine/commit/5f0271a38988732c48f55c1244717430816391d3) for this from @blasten is in the 2.12 beta. (@blasten Is that the right commit?) Please give it a try.

**Comment by marcellocamara** [comment]
> is in the 2.12 beta.

I've tested on `2.12.0-4.1.pre` with dart `2.17.0-182.1.beta` using PageView with BannerAd inside pages and the bug still exists. 😢 
Can anyone test it to confirm too ?

**Comment by blasten** [comment]
The ultimate fix for this issue is available on the master channel. @marcellocamara can you try building with the master channel?

**Comment by blasten** [comment]
Thanks for confirming. 

If someone is having this issue. please try the steps described above, and comment
if you are having issues after trying.
